### PR TITLE
Switch to multi module build

### PR DESCRIPTION
**Pull request by simonsouter**
Move test classes to separate project.
Add sbt-release plugin

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,57 +1,40 @@
...
+  </scm>
+    <developers>
+      <developer>
+        <id>simon</id>
+        <name>Simon Souter</name>
+        <url>https://github.com/simonsouter</url>
+      </developer>
+    </developers>,
+
+  licenses := ("GNU General Public License (GPL), Version 3.0", url("http://opensource.org/licenses/GPL-3.0")) :: Nil
```
Comment (jkpl) [inline build.sbt]: License change?

**Comment by SohumB** [comment]
:+1:

File: build.sbt (hunk 1 above)
Comment (simonsouter) [inline build.sbt]: Yes, as instructed my Dr Squire...

Comment (jkpl) [inline build.sbt]: As far as I know, code using GPL licensed library has to be licensed under GPL as well. Thus everything using the library would also need to share their code out in the public.

Comment (jkpl) [inline build.sbt]: I believe LGPL allows using the library without having to share the code. https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License

**Pull request merged**